actors/miner/src/lib.rs

```diff
@@ -3744,33 +3755,48 @@ fn validate_extended_expiration(
 }
 
 fn extend_simple_qap_sector(
+    policy: &Policy,
     new_expiration: ChainEpoch,
+    curr_epoch: ChainEpoch,
     sector: &SectorOnChainInfo,
-    claim_space_by_sector: &BTreeMap<SectorNumber, u64>,
+    claim_space_by_sector: &BTreeMap<SectorNumber, (u64, u64)>,
 ) -> Result<SectorOnChainInfo, ActorError> {
     let mut new_sector = sector.clone();
     if sector.verified_deal_weight > BigInt::zero() {
         let old_duration = sector.expiration - sector.activation;
         let deal_space = &sector.deal_weight / old_duration;
-        let verified_deal_space = &sector.verified_deal_weight / old_duration;
-        let expected_verified_deal_space = match claim_space_by_sector.get(&sector.sector_number) {
-            None => {
-                return Err(actor_error!(
-                    illegal_argument,
-                    "claim missing from declaration for sector {}",
-                    sector.sector_number
-                ))
-            }
-            Some(space) => space,
-        };
-        if BigInt::from(*expected_verified_deal_space as i64) != verified_deal_space {
-            return Err(actor_error!(illegal_argument, "declared verified deal space in claims ({}) does not match verified deal space ({}) for sector {}", expected_verified_deal_space, verified_deal_space, sector.sector_number));
+        let old_verified_deal_space = &sector.verified_deal_weight / old_duration;
+        let (expected_verified_deal_space, new_verified_deal_space) =
+            match claim_space_by_sector.get(&sector.sector_number) {
+                None => {
+                    return Err(actor_error!(
+                        illegal_argument,
+                        "claim missing from declaration for sector {}",
+                        sector.sector_number
+                    ))
+                }
+                Some(space) => space,
+            };
+        // claims must be completely accounted for
+        if BigInt::from(*expected_verified_deal_space as i64) != old_verified_deal_space {
+            return Err(actor_error!(illegal_argument, "declared verified deal space in claims ({}) does not match verified deal space ({}) for sector {}", expected_verified_deal_space, old_verified_deal_space, sector.sector_number));
         }
+        // claim dropping is restricted to extensions at the end of a sector's life
+
+        let dropping_claims = expected_verified_deal_space != new_verified_deal_space;
+        if dropping_claims && sector.expiration - curr_epoch >= policy.end_of_life_claim_drop_period
+        {
+            return Err(actor_error!(
+                forbidden,
+                "attempt to drop sectors with {} epochs < end of life claim drop period remaining"
```

Comment:
Missing format argument. Also include policy.end_of_life_claim_drop_period in the message